emf/src/main/kotlin/com/strumenta/kolasu/emf/Metamodel.kt

```diff
@@ -290,33 +216,29 @@ class MetamodelBuilder(packageName: String, nsURI: String, nsPrefix: String, res
         eClass.isAbstract = kClass.isAbstract || kClass.isSealed
         eClass.isInterface = kClass.java.isInterface
 
+        kClass.typeParameters.forEach { kTypeParameter: KTypeParameter ->
+            eClass.eTypeParameters.add(
+                EcoreFactory.eINSTANCE.createETypeParameter().apply {
+                    name = kTypeParameter.name
+                }
+            )
+        }
```

Comment:
This looks good, just a comment.

We do not support bounds, e.g. A<B :  C>, and more complex specifications involving generics, e.g. A<D, B: C<D>>. I think we should keep track of this, in case it turns out to be required (not sure). 

Ideally, this could be achieved processing `kTypeParamater.upperBounds` and map these into `EGenericType` instances (similarly to those created for references).

Reply:
Right, we should add a comment. For implementing support however we should consider that  Kotlin has this concept of  out and in variant which I do not think map exactly to things in EMF, so I am not sure we can support them properly.

Reply:
I agree, maybe the variance information could be annotated using an EAnnotation.

Reply:
Good idea!